tests/components/websocket_api/test_messages.py

```diff
@@ -0,0 +1,65 @@
+"""Test Websocket API messages module."""
+
+from homeassistant.components.websocket_api.messages import (
+    cached_event_message,
+    message_to_json,
+)
+from homeassistant.const import EVENT_STATE_CHANGED
+from homeassistant.core import callback
+
+
+async def test_cached_event_message(hass):
+    """Test that we cache event messages."""
+
+    events = []
+
+    @callback
+    def _event_listener(event):
+        events.append(event)
+
+    hass.bus.async_listen(EVENT_STATE_CHANGED, _event_listener)
+
+    hass.states.async_set("light.window", "on")
+    hass.states.async_set("light.window", "off")
+    await hass.async_block_till_done()
+
+    assert len(events) == 2
+
+    msg0 = cached_event_message(2, events[0])
+    assert msg0 == cached_event_message(2, events[0])
+
+    msg1 = cached_event_message(2, events[1])
+    assert msg1 == cached_event_message(2, events[1])
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6212145)